src/main/java/edu/cornell/kfs/gl/businessobject/lookup/CuCurrentAccountBalanceLookupableHelperServiceImpl.java

```diff
@@ -0,0 +1,269 @@
+     * of the updateCurrentBalance() method.
+     */
+    @SuppressWarnings("deprecation")
+    protected class BalanceUpdater {
+
+        protected final CurrentAccountBalance currentBalance;
+        protected final Balance balance;
+        protected final String fiscalPeriod;
+        protected final Collection<String> cashBudgetRecordLevelCodes;
+        protected final Collection<String> expenseObjectTypeCodes;
+        protected final Collection<String> fundBalanceObjCodes;
+        protected final Collection<String> currentAssetObjTypeCodes;
+        protected final Collection<String> currentLiabilityObjTypeCodes;
+        protected final Collection<String> incomeObjTypeCodes;
+        protected final Collection<String> encumbranceBalTypes;
+        protected final Collection<String> assetLiabilityFundBalanceTypeCodes;
+        protected final boolean excludeCBPeriod;
+        protected final String balanceTypeCode;
+        protected final String objectTypeCode;
+        protected final String objectCode;
+        protected final Account replacementAccount;
+        protected final boolean isCashBudgetRecording;
+        
+        public BalanceUpdater(CurrentAccountBalance currentBalance, Balance balance, String fiscalPeriod) {
+            this.currentBalance = currentBalance;
+            this.balance = balance;
+            this.fiscalPeriod = fiscalPeriod;
+            
+            this.cashBudgetRecordLevelCodes = getParameterValuesAsString(KFSParameterKeyConstants.GlParameterConstants.CASH_BUDGET_RECORD_LEVEL);
+            this.expenseObjectTypeCodes = getParameterValuesAsString(KFSParameterKeyConstants.GlParameterConstants.EXPENSE_OBJECT_TYPE);
+            this.fundBalanceObjCodes = getParameterValuesAsString(KFSParameterKeyConstants.GlParameterConstants.FUND_BALANCE_OBJECT_CODE);
+            this.currentAssetObjTypeCodes = getParameterValuesAsString(CUKFSParameterKeyConstants.GlParameterConstants.CURRENT_ASSET_OBJECT_TYPE_CODE);
+            this.currentLiabilityObjTypeCodes = getParameterValuesAsString(CUKFSParameterKeyConstants.GlParameterConstants.CURRENT_LIABILITY_OBJECT_TYPE_CODE);
+            this.incomeObjTypeCodes = getParameterValuesAsString(KFSParameterKeyConstants.GlParameterConstants.INCOME_OBJECT_TYPE);
+            this.encumbranceBalTypes = getParameterValuesAsString(KFSParameterKeyConstants.GlParameterConstants.ENCUMBRANCE_BALANCE_TYPE);
+            this.excludeCBPeriod = getParameterValueAsBoolean(CUKFSParameterKeyConstants.GlParameterConstants.EXCLUDE_CB_PERIOD, Boolean.FALSE)
+                    .booleanValue();
+            
+            SystemOptions options = getOptionsService().getCurrentYearOptions();
+            this.assetLiabilityFundBalanceTypeCodes = Arrays.asList(options.getFinancialObjectTypeAssetsCd(),  // AS
+                options.getFinObjectTypeLiabilitiesCode(), // LI
+                options.getFinObjectTypeFundBalanceCd());  // FB

```

Comment:
Those were in the base KFS code, but I'll remove them for the sake of cleanliness.